packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -79,11 +79,13 @@ export default class AssetGraphBuilder extends EventEmitter {
     this.options = options;
     this.assetRequests = [];
 
-    let {hot} = options;
+    // TODO: changing these should not throw away the entire graph.
+    // We just need to re-run target resolution.
+    let {hot, publicUrl, distDir, minify, scopeHoist} = options;
```

Comment:
Should we have a node in the request graph to invalidate based on options? @padmaia

Reply:
Yeah, that sounds like a good idea.